src/acquisition/covidcast/database.py

```diff
@@ -247,6 +246,104 @@ def insert_or_update_batch(self, cc_rows, batch_size=2**20, commit_partial=False
       self._cursor.execute(drop_tmp_table_sql)
     return total
 
+  def delete_batch(self, cc_deletions):
+    """
+    Remove rows specified by a csv file or list of tuples.
+
+    If cc_deletions is a filename, the file should include a header row and use the following field order:
+    - geo_id
+    - value (ignored)
+    - stderr (ignored)
+    - sample_size (ignored)
+    - issue (YYYYMMDD format)
+    - time_value (YYYYMMDD format)
+    - geo_type
+    - signal
+    - source
+
+    If cc_deletions is a list of tuples, the tuples should use the following field order (=same as above, plus time_type):
+    - geo_id
+    - value (ignored)
+    - stderr (ignored)
+    - sample_size (ignored)
+    - issue (YYYYMMDD format)
+    - time_value (YYYYMMDD format)
+    - geo_type
+    - signal
+    - source
+    - time_type
+    """
+    tmp_table_name = "tmp_delete_table"
+    create_tmp_table_sql = f'''
+CREATE OR REPLACE TABLE {tmp_table_name} LIKE covidcast;
+'''
+
+    amend_tmp_table_sql = f'''
+ALTER TABLE {tmp_table_name} ADD COLUMN covidcast_id bigint unsigned;
+'''
+
+    load_tmp_table_infile_sql = f'''
+LOAD DATA INFILE "{cc_deletions}"
+INTO TABLE {tmp_table_name}
+FIELDS TERMINATED BY ","
+IGNORE 1 LINES
+(`geo_value`, `value`, `stderr`, `sample_size`, `issue`, `time_value`, `geo_type`, `signal`, `source`)
+SET time_type="day";
+'''
+
+    load_tmp_table_insert_sql = f'''
+INSERT INTO {tmp_table_name}
+(`geo_value`, `value`, `stderr`, `sample_size`, `issue`, `time_value`, `geo_type`, `signal`, `source`, `time_type`,
+`value_updated_timestamp`, `direction_updated_timestamp`, `lag`, `direction`, `is_latest_issue`)
+VALUES
+(%s, %s, %s, %s, %s, %s, %s, %s, %s, %s,
+0, 0, 0, 0, 0)
+'''
+
+    add_id_sql = f'''
+UPDATE {tmp_table_name} d INNER JOIN covidcast c USING
+(`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`, `issue`)
+SET d.covidcast_id=c.id, d.is_latest_issue=c.is_latest_issue;
+'''
+
+    delete_sql = f'''
+DELETE c FROM {tmp_table_name} d INNER JOIN covidcast c WHERE d.covidcast_id=c.id;
+'''
+
+    fix_latest_issue_sql = f'''
+UPDATE
```

Comment:
This is probably just some syntax thing I'm unfamiliar with it, but it certainly reads like the update statement is acting on the subquery. Is it just updating both tables at once?

If possible, it might be more readable if it was restructured as:
```
UPDATE `covidcast`
SET `is_latest_issue`=1
FROM
<subquery> LEFT JOIN  `covidcast` USING
```

Reply:
Oops sorry for late reply. Yea that makes sense, the SET qualification sounds good to me